pkg/receive/config.go

```diff
@@ -67,7 +68,7 @@ type HashringConfig struct {
 	Tenants        []string          `json:"tenants,omitempty"`
 	Endpoints      []Endpoint        `json:"endpoints"`
 	Algorithm      HashringAlgorithm `json:"algorithm,omitempty"`
-	ExternalLabels map[string]string `json:"external_labels,omitempty"`
+	ExternalLabels labels.Labels     `json:"external_labels,omitempty"`
```

Comment:
Shouldn't this continue to be `map[string]string`? When `labels.Labels` isn't `map[string]string` then this will break, no?

Reply:
labels.Labels implements json.Unmarshaler and calls essentially "FromMap": https://github.com/prometheus/prometheus/blob/bdc3cfdd5dff0e076e1a5778289b36b3fe5962f3/model/labels/labels_common.go#L73 so i think it should work ( with added benefit of validation )